src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
+    end
+  end
+
+  def select!(&block : K, V -> U)
+    delete_if{|k, v| !yield k, v }
```

Comment:
I just link https://gist.github.com/jhass/a5ae80d87f18e53e7b56 instead of saying what I think of that one ... :P